src/classes/site.h

```diff
@@ -24,12 +26,20 @@ class Site
      * Site Definition
      */
     protected:
+    // Site definition used to generate site
+    const SpeciesSite *parent_;
+    // Unique site index in the parent
+    int uniqueSiteIndex_;
```

Comment:
An `optional` would perhaps be a better choice, and remove the need for the magic number `-1`?
```suggestion
    std::optional<int> uniqueSiteIndex_;
```